security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -447,10 +457,28 @@ func (sc *SecretCache) generateSecret(ctx context.Context, token, resourceName s
 		return nil, err
 	}
 
-	certChainPEM, err := sc.fetcher.CaClient.CSRSign(ctx, csrPEM, exchangedToken, int64(sc.configOptions.SecretTTL.Seconds()))
-	if err != nil {
-		log.Errorf("Failed to sign cert for %q: %v", resourceName, err)
-		return nil, err
+	startTime := time.Now()
+	retry := 0
+	backOffInMilliSec := initialBackOffIntervalInMilliSec
+	var certChainPEM []string
+	for true {
+		certChainPEM, err = sc.fetcher.CaClient.CSRSign(ctx, csrPEM, exchangedToken, int64(sc.configOptions.SecretTTL.Seconds()))
+		if err == nil {
+			break
+		}
+
+		isTimeout := startTime.Add(time.Millisecond * envoyDefaultTimeoutInMilliSec).Before(time.Now())
+		// If non-retryable error or reach envoy timeout, fail the request by returning err
+		if !isRetryableErr(status.Code(err)) || isTimeout {
```

Comment:
The logic in this not accurate in this if block as we are catching non-retryable error or timeout, however, we never log/return timeout error.

Reply:
updated.